server/src/main/java/org/opensearch/action/support/StreamSearchChannelListener.java

```diff
@@ -68,8 +68,14 @@ public final void onResponse(Response response) {
     public void onFailure(Exception e) {
         try {
             channel.sendResponse(e);
+            // Don't call completeStream() here as sendResponse() already releases the channel
         } catch (IOException exc) {
-            channel.completeStream();
+            // If sendResponse fails, try to complete the stream
+            try {
+                channel.completeStream();
```

Comment:
I'm skeptical about this change because of nature of this API. `completeStream()` always indicate successful completion whereas sendResponse(Exception) is the one to be used in case of errors. 

This is what documentation says from `TransportChannel`- 
```
     * Sends a batch of responses to the request that this channel is associated with.
     * Call {@link #completeStream()} on a successful completion.
     * For errors, use {@link #sendResponse(Exception)} and do not call {@link #completeStream()}
```

I think calling completeStream here was a miss earlier, so we can remove this safely and just log a warning.